packages/@react-spectrum/table/src/Table.tsx

```diff
@@ -241,7 +241,14 @@ function TableCollectionView({layout, collection, focusedKey, renderView, render
     transitionDuration: collection.body.props.isLoading && collection.size > 0 ? 0 : 500
   });
 
-  let {collectionViewProps} = useCollectionView({focusedKey}, collectionState, domRef);
+  let {collectionViewProps} = useCollectionView({
+    focusedKey,
+    shouldScrollToItem(key) {
+      // Prevent scrolling to the top when clicking on column headers.
+      let item = collection.getItem(key);
+      return item?.type !== 'column';
+    }
```

Comment:
Noticed that a user can no longer cause the table to scroll horizontally when navigating through the headers via the left/right arrow keys. Was this also a expected change or a side effect of blocking scrolling when clicking on column headers (perhaps this func should detect what action caused the scroll to happen?)

Reply:
oo, yep, i can reproduce, womp womp

Reply:
Adjusted this to only scroll in the X direction when focusing column headers. Also improved horizontal scrolling with the keyboard to offset by the width of the selection column, which would otherwise overlap the selected item.